lib/awful/titlebar.lua

```diff
@@ -447,6 +459,51 @@ local titlebar = {
 -- @tparam gears.surface|string path
 -- @see gears.surface
 
+--- Add tooltip texts for titlebar buttons: active, inactive, plain
+-- @beautiful beautiful.titlebar_tooltip_messages
+-- @tparam table tooltips Key-Value syntax: key is button name, value is tooltip text.
+-- See default usage in variable `default_tooltip_messages` in titlebar component.
+-- @see awful.titlebar
+-- @usage -- Configure titlebar button tooltip texts in your theme.
+-- theme.titlebar_tooltip_messages = {
+--     close = "Close",
+--     minimize = "Minimize",
+--     maximized_active = "Unmaximize",
+--     maximized_inactive = "Maximize",
+--     floating_active = "Tiling",
+--     floating_inactive = "Floating",
+--     ontop_active = "NotOnTop",
+--     ontop_inactive = "OnTop",
+--     sticky_active = "NotSticky",
+--     sticky_inactive = "Sticky"
+-- }
```

Comment:
Yes. This was previous state. Elv13 had same notes, you are in same team :)
So, theme variables are generated by your rules: "titlebar_" .. name .. "_button_" .. prefix .. img .. state. Just check last modifs driven by elv13.

Reply:
I don't think it's related to @Elv13 comments.

I ask to not use a table, but multiple different string variables. So that instead of `theme.titlebar_tooltip_messages.<variable>_<status>` we would have `theme.titlebar_tooltip_messages_<variable>_<status>`. It would make it more consistent with the existing code base.

Reply:
Sorry if I wasn't clear before. Yes, the names should ideally be flat (no sub-tables). And yes, that's pretty annoying for everybody, but rather hard to fix at this point. I would prefer the names to remain flat as @Aire-One mentions.

The historical reason it is like this is just sad[1], but to break a convention, you have to have a really god reason. Good consistency really helps lower the learning curve.

[1] [Well into the 3.x branch](https://github.com/awesomeWM/awesome/blob/3.1/themes/default/theme.in), the themes were `rc` files rather than `lua` files. This was a leftover from the 1.x/2.x days where Awesome scripting was actually implemented over an RPC in Ruby. To easy the transition (now like 14 years old), the names were kept so the `rc` file could be renamed `lua` with `theme.` added in front of each line,

Reply:
Done. Doc is updated according to this recommendation.